frontend/src/components/Nav.js

```diff
@@ -0,0 +1,131 @@
+import React, { useState, useEffect } from "react";
+import styled from "styled-components";
+import { LIGHT_PURPLE, PURPLE } from "../constants/colors";
+import { MenuOutlined } from "@ant-design/icons";
+
+const NavBar = styled.div`
+  background-color: ${LIGHT_PURPLE};
+  display: flex;
+  height: 50px;
+  justify-content: space-between;
+  position: fixed;
+  width: 100%;
+`;
+
+const NavContainer = styled.div`
+  align-items: center;
+  display: flex;
+  height: 100%;
+  margin-right: 2em;
+
+  a:not(:last-child) {
+    margin-right: 1.5em;
+  }
+`;
+
+const ProjectName = styled.div`
+  align-items: center;
+  display: flex;
+  color: ${PURPLE};
+  font-weight: 600;
+  margin-left: 2em;
+  text-transform: uppercase;
+`;
+
+const Link = styled.a`
+  color: ${PURPLE};
+  text-decoration: none;
+  font-weight: 600;
+  ${(props) =>
+    props.size &&
+    `
+    font-size: ${props.size}px;
+  `}
+`;
+
+const StyledMenuIcon = styled(MenuOutlined)`
+  align-items: center;
+  color: ${PURPLE};
+  display: flex;
+  font-size: 20px;
+  margin-right: 1.5em;
+
+  :hover {
+    cursor: pointer;
+    transition: all 0.2s ease-in-out;
+    transform: scale(1.5);
+  }
+`;
+
+const OpenMenu = styled.div`
+  background-color: ${LIGHT_PURPLE};
+  height: 50vh;
+  margin-top: 40px;
+  position: fixed;
+  width: 100%;
+`;
+
+const OpenMenuContainer = styled.div`
+  align-items: center;
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+  justify-content: center;
+
+  a:not(:last-child) {
+    margin-bottom: 2em;
+  }
+`;
+
+const Nav = () => {
+  const [screenWidth, setScreenWidth] = useState(window.innerWidth);
+  const [isOpen, setIsOpen] = useState(false);
+  useEffect(() => {
+    const resizeScreen = () => {
+      setScreenWidth(window.innerWidth);
+    };
+    window.addEventListener("resize", resizeScreen);
+    return () => {
+      window.removeEventListener("resize", resizeScreen);
+    };
+  }, []);
+
+  const HeaderItems = (s) => {
+    return (
+      <>
```

Comment:
What does this do? I've never seen it before.

Reply:
Ohh it's like when you don't wanna make another parent div you just use an empty <></> to wrap around all your child components